### DISCUSSION TITLE
Footnotes in title block in HTML

### DISCUSSION THREAD

**Discussion by twsh** [jgm/pandoc#9426] [Q&A]
I am trying to use footnotes in a title block for things like acknowledgements and author affiliation. The following works fine for PDF, LaTeX, ODT:

```
% Title[^a]
% Author[^b]

[^a]: A
[^b]: B
```

With HTML, the footnote on the title is numbered 2, and the one on the author is numbered 1.

Is there a reason why it should/must be that way? 

https://pandoc.org/try/?params=%7B%22text%22%3A%22%25+Title%5B%5Ea%5D%5Cn%25+Author%5B%5Eb%5D%5Cn%5Cn%5B%5Ea%5D%3A+A%5Cn%5B%5Eb%5D%3A+B%22%2C%22to%22%3A%22html5%22%2C%22from%22%3A%22markdown%22%2C%22standalone%22%3Atrue%2C%22embed-resources%22%3Afalse%2C%22table-of-contents%22%3Afalse%2C%22number-sections%22%3Afalse%2C%22citeproc%22%3Afalse%2C%22html-math-method%22%3A%22plain%22%2C%22wrap%22%3A%22auto%22%2C%22highlight-style%22%3Anull%2C%22files%22%3A%7B%7D%2C%22template%22%3Anull%7D 

This is what the title block is now:

```
<header id="title-block-header">
<h1 class="title">Title<a href="#fn2" class="footnote-ref" id="fnref2"
role="doc-noteref"><sup>2</sup></a></h1>
<p class="author">Author<a href="#fn1" class="footnote-ref" id="fnref1"
role="doc-noteref"><sup>1</sup></a></p>
</header>
```

This is what it , presumably, should be:

```
<header id="title-block-header">
<h1 class="title">Title<a href="#fn1" class="footnote-ref" id="fnref1"
role="doc-noteref"><sup>1</sup></a></h1>
<p class="author">Author<a href="#fn2" class="footnote-ref" id="fnref2"
role="doc-noteref"><sup>2</sup></a></p>
</header>
```

Pandoc version 3.1.11.1

**Comment by jgm** [comment]
I suspect the answer is this.  The metadata fields are stored in a Data.Map.Map, and the map is not an ordered container, so the order in which the fields are parsed by the writer in populating template variables is not determinate. Unfortunately, this order affects the footnote order.

I suppose we could try altering `metaToContext` to ensure that it processes the `title` field before the others.

**Reply by jgm** [reply]
The reason you don't see this in PDF is that there, LaTeX handles the numbering, not our LaTeX writer.

**Reply by jgm** [reply]
If you want to submit an issue, the relevant bit of code is https://github.com/jgm/pandoc/blob/main/src/Text/Pandoc/Writers/Shared.hs#L96-L103

**Reply by twsh** [reply]
@jgm Thanks. I played around with a few things and I'm pretty sure that they end up numbered in alphabetical order of metadata field (author < date < subtitle < title). The thing that makes me hesitate about raising an issue is that I'm not sure what a good solution would be. Making title always first wouldn't solve the issue with a footnote in a subtitle and an author field, for example. But would it be a good idea to force Pandoc to process them in a specific order, corresponding to the order in the HTML template? What if someone has a template where they are arranged differently? Maybe someone will think of a neat solution.

**Reply by jgm** [reply]
> corresponding to the order in the HTML template?

We don't currently have that information.  (Though doctemplates could be modified to provide it.)
